app/modules/ws2812.c

```diff
@@ -165,6 +165,110 @@ static int ws2812_buffer_fade(lua_State* L) {
   return 0;
 }
 
+static int ws2812_buffer_fade_in(lua_State* L) {
+  ws2812_buffer * buffer = (ws2812_buffer*)lua_touserdata(L, 1);
+  const int fade = luaL_checkinteger(L, 2);
+
+  luaL_argcheck(L, buffer && buffer->canary == CANARY_VALUE, 1, "ws2812.buffer expected");
+  luaL_argcheck(L, fade > 0, 2, "fade value should be a strict positive number");
+
+  uint8_t * p = &buffer->values[0];
+  int i;
+  uint8_t val = 0;
+  for(i = 0; i < buffer->size * buffer->colorsPerLed; i++)
+  {
+    val = *p * fade;
+    if (val > 255) val = 255;
+    *p++ = val;
+  }
+
+  return 0;
+}
+
+
+static int ws2812_buffer_shift(lua_State* L) {
+  ws2812_buffer * buffer = (ws2812_buffer*)lua_touserdata(L, 1);
+  const int shiftValue = luaL_checkinteger(L, 2);
+
+  luaL_argcheck(L, buffer && buffer->canary == CANARY_VALUE, 1, "ws2812.buffer expected");
+  luaL_argcheck(L, shiftValue > 0-buffer->size && shiftValue < buffer->size, 2, "shifting more elements than buffer size");
+
+  int shift = shiftValue >= 0 ? shiftValue : -shiftValue;
+    
+    
+  // check if we want to shift at all
+  if (shift == 0)
+  {
+    return 0;
+  }
+  
+
+  int * tmp_pixels = luaM_malloc(L, buffer->colorsPerLed * sizeof(int) * shift);
```

Comment:
Yes, I did this because luaL_checkinteger returns int. I could have cast luaL_checkinteger to uint8_t it there and win some cycles and few memory bytes, should check the assembly :)

But here, no need of (int), uint8_t is enough as it's your input and output type